packages/flutter_tools/test/src/base/file_system_test.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('file_system', () {
```

Comment:
You could instantiate a second `MemoryFileSystem` for the destination directory and make sure there's no cross-pollution between the two file systems.